### PR TITLE
Add and use supports_celery attribute for cli celery command validation

### PR DESCRIPTION
**Pull request by alexchen8**
Replacing existing celery subclass check in celery cli validation with a new attribute to Celery based executors.
Also moved celery module check a level higher. IMO, it is worth checking even if the user is using built-in Celery or C-K executors.

The reason behind this change is to allow any custom executor to utilize built-in airflow celery executor/worker. There are some cases that doesn't make sense to subclass the existing Celery / C-K executors (for example a similar implementation of C-K executor that has some other executors for routing, but not Kubernetes").

This change should be backward compatible, assuming subclasses of C/C-K in the wild does not have `supports_celery` attribute defined for other use.

### CODE REVIEW

**Comment by ac1997** [comment]
@ashb Just want to see what's your thought on this as the last modification was from you.

**Comment by ac1997** [comment]
Just want to resurface this PR... @potiuk Do you have any thoughts on this?

File: airflow/cli/cli_parser.py
Comment (potiuk) [inline airflow/cli/cli_parser.py]:
```suggestion
                        f"celery subcommand works only with executors that have 'supports_celery' "
```

File: tests/cli/test_cli_parser.py
Comment (potiuk) [inline tests/cli/test_cli_parser.py]:
```suggestion
            "works only with executors that have 'supports_celery' set to True, "
```

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
I will need one more committer's approval though.

**Comment by ac1997** [comment]
My apologies, just noticed that my initial commit was done using company email address. Had to rebase/force push to use personal email.

**Comment by potiuk** [comment]
Anyone else?

**Comment by ac1997** [comment]
Bumping this PR for review

**Pull request closed without merging**
